lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -417,6 +515,72 @@ void ScienceSensorsSystemPrivate::ReadData()
   this->initialized = true;
 }
 
+/////////////////////////////////////////////////
+void ScienceSensorsSystemPrivate::UpdateWorldSphericalOrigin(
+  ignition::gazebo::EntityComponentManager &_ecm)
+{
+  // Lock for changes to worldOriginSpherical* until update complete
+  std::lock_guard<std::mutex> lock(mtx);
+
+  if (!this->worldSphericalCoordsInitialized)
```

Comment:
Actually, we cannot return right away. That's actually a logic error due to incorrect braces. Thanks for catching that.

The if-statement is there because `UpdateWorldSphericalOrigin()` is called from two places, one in `Configure()` (because otherwise no data is shown), and the other in `PreUpdate()`. In the former case, the variable `worldSphericalCoordsInitialized` is false, and in the latter case, it is true. BUT, in both cases, `ShiftDataToNewSphericalOrigin()` needs to be called. So the `ShiftDataToNewSphericalOrigin()` call is supposed to go outside the if-statement.

That hadn't been caught because `ShiftDataToNewSphericalOrigin()` is not coded yet! So both cases the behavior is the same.

Reply:
I pulled the function call out of the if-statement (to be committed, will post hash), but the logic might need to be changed once the function is coded. The logic is a combination of `this->worldSphericalCoordsInitialized` and `this->initialized`, which needs to be thought about and tested after `ShiftDataToNewSphericalOrigin()` is implemented.

Really, I just put the calls in there for whoever that implements `ShiftDataToNewSphericalOrigin()` to be aware of these different circumstances under which `UpdateWorldSphericalOrigin()` may be called. The call to `UpdateWorldSphericalOrigin()` in `PreUpdate()` needs to be tested at the same time too, because that condition has also never happened so far. It only happens if a vehicle is spawned.

Reply:
TODOs added in c287169